### PR TITLE
First draft of idea

### PR DESCRIPTION
**Pull request by dhardiman**
@drhaynes This is my idea for simple http fetches in swift using generics to try and eliminate boilerplate code. Would appreciate thoughts, suggestions, changes. If you think it looks a good idea I'll tidy it up properly, maybe get Roman and Steve to review as well.

### CODE REVIEW

File: Example/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+//  Example
+//
+//  Created by David Hardiman on 13/02/2016.
+//  Copyright © 2016 David Hardiman. All rights reserved.
+//
+
+import UIKit
+import Fetch
+
+class ViewController: UIViewController {
```
Comment (dhardiman) [inline Example/ViewController.swift]: This demonstrates how to use it. No magic, just create a model object that knows how to parse the response and pass the type to `get`

File: Fetch/Fetch.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ The result of a fetch request
+ 
+ - Successful: The fetch and parse was successful
+ - Failure:    The fetch returned, but the object returned represents a failure
+ - Error:      The fetch failed with an error
+ */
+public enum FetchResult<T: Parsable> {
+    case Successful(T)
+    case Failure(T)
```
Comment (dhardiman) [inline Fetch/Fetch.swift]: Maybe this case should also return an `ErrorType` that `Parsable` is responsible for generating

File: Fetch/Parsable.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+//  Created by David Hardiman on 13/02/2016.
+//  Copyright © 2016 David Hardiman. All rights reserved.
+//
+
+import Foundation
+
+/**
+ *  Item that can parse data retrieved from a request
+ */
+public protocol Parsable {
```
Comment (dhardiman) [inline Fetch/Parsable.swift]: Not convinced by this as a name. Perhaps something else might make more sense

File: Fetch/Request.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+//  Created by David Hardiman on 13/02/2016.
+//  Copyright © 2016 David Hardiman. All rights reserved.
+//
+
+import Foundation
+
+/**
+ *  Simple struct representing a network request
+ */
+public struct Request {
```
Comment (dhardiman) [inline Fetch/Request.swift]: We could just use `NSURLRequest` directly, but it's a bit manky from swift, especially as you have to use the mutable alternative. Seemed nice to hide it behind pure swift.

**Comment by dhardiman** [comment]
- [ ] Confirm name of `Parsable`
- [ ] Fix return queue for fetch (currently arbitrary, should be main queue by default but allow customisation)
- [ ] Is `get`, `post` the correct method names?
- [ ] Add `delete` method? This is why I think method names might need to change
- [ ] Add tests
- [ ] Add readme
- [ ] Add license

File: Fetch/Request.swift (hunk 4 above)
Comment (drhaynes) [inline Fetch/Request.swift]: That's much more Swifty, very nice.

**Pull request merged**